### PR TITLE
Update test_unicodedata from CPython 3.11.2

### PR DESCRIPTION
**Pull request by dalinaum**
#4564

### CODE REVIEW

File: Lib/test/test_unicodedata.py [hunk 1]
```diff
@@ -63,33 +61,24 @@ def test_method_checksum(self):
...
 
 class UnicodeFunctionsTest(UnicodeDatabaseTest):
 
     # Update this if the database changes. Make sure to do a full rebuild
     # (e.g. 'make distclean && make') to get the correct checksum.
-    expectedchecksum = 'c44a49ca7c5cb6441640fe174ede604b45028652'
-    # TODO: RUSTPYTHON
-    @unittest.expectedFailure
+    expectedchecksum = '98d602e1f69d5c5bb8a5910c40bbbad4e18e8370'
+    @requires_resource('cpu')
```
Comment (fanninpm) [inline Lib/test/test_unicodedata.py:76]: I would keep the `# TODO: RUSTPYTHON` and `@unittest.expectedFailure` in one commit, and remove it (if necessary) in a second commit.

Comment (dalinaum) [reply Lib/test/test_unicodedata.py:76]: @fanninpm I just seprated it into two commits.

Comment (dalinaum) [reply Lib/test/test_unicodedata.py:76]: @fanninpm The test didn't succeed. I would revert back to removing `@unittest.expectedFailure`.

**Review by youknowone — APPROVED** [review]
Thank you!

**Pull request merged**